### PR TITLE
152 first run of 2024 robot

### PR DESCRIPTION
**Pull request by gdoffe**
Batch of set and improvments to make the 2024 robot almost functional.

### CODE REVIEW

**Review by ecourtois — APPROVED** [review]
LGTM after the few pending fixes.

File: motion_control/filters/pose_straight_filter/PoseStraightFilter.cpp [hunk 1]
```diff
@@ -135,17 +138,21 @@ void PoseStraightFilter::execute() {
     // Linear current speed
     outputs_[1] = current_speed.distance();
     // Linear target speed
-    outputs_[2] = target_speed.distance();
+    outputs_[2] = fabs(target_speed.distance());
+    // Should linear speed be filtered ?
```
[the remark this answers is not in the document]
Comment (gdoffe) [reply motion_control/filters/pose_straight_filter/PoseStraightFilter.cpp]: done

File: motion_control/filters/pose_straight_filter/include/pose_straight_filter/PoseStraightFilterParameters.hpp [hunk 2]
```diff
@@ -23,38 +23,60 @@ class PoseStraightFilterParameters {
...
+    /// Set angular threshold
+    void set_intermediate_angular_threshold(
+        double angular_intermediate_threshold                            ///< [in]   intermediate angular threshold
+        ) { angular_intermediate_threshold_ = angular_intermediate_threshold; };
+
+    ///  Get angular deceleration
+    /// return Angular deceleration
+    double angular_deceleration() const { return angular_deceleration_; };
+
+    ///  Get linear deceleration
```
[the remark this answers is not in the document]
Comment (gdoffe) [reply motion_control/filters/pose_straight_filter/include/pose_straight_filter/PoseStraightFilterParameters.hpp]: done

File: motion_control/filters/pose_straight_filter/PoseStraightFilter.cpp [hunk 3]
```diff
@@ -135,17 +138,21 @@ void PoseStraightFilter::execute() {
...
     // Angular pose error
-    outputs_[3] = pos_err.angle();
+    outputs_[4] = pos_err.angle();
     // Angular current speed
-    outputs_[4] = current_speed.angle();
+    outputs_[5] = current_speed.angle();
     // Angular target speed
-    outputs_[5] = target_speed.angle();
+    outputs_[6] = fabs(target_speed.angle());
+    // Should angular speed be filtered ?
```
[the remark this answers is not in the document]
Comment (gdoffe) [reply motion_control/filters/pose_straight_filter/PoseStraightFilter.cpp]: done

File: Makefile.include [hunk 4]
```diff
@@ -20,6 +20,7 @@ QUIET ?= 1
 DEVELHELP ?= 1
 THREAD_NAMES = 1
 
+REMOTE_USER ?= pi
```
[the remark this answers is not in the document]
Comment (gdoffe) [reply Makefile.include]: done

File: boards/cogip-board/Makefile.dep [hunk 5]
```diff
@@ -2,4 +2,7 @@ ifneq (,$(filter stdio_uart,$(USEMODULE)))
   FEATURES_REQUIRED += periph_lpuart
 endif
 
+# Disable automatic init of I2C as some board have different IOs mapping
```
[the remark this answers is not in the document]
Comment (gdoffe) [reply boards/cogip-board/Makefile.dep]: done

File: platforms/pf-robot-motion-control/motion_control.cpp [hunk 6]
```diff
@@ -427,14 +440,33 @@ void pf_motor_drive(const cogip::cogip_defs::Polar &command)
         // Limit commands to what the PWM driver can accept as input in the range [INT16_MIN:INT16_MAX].
         // The PWM driver will filter the value to the max PWM resolution defined for the board.
         // Compute motor commands with Polar motion control result
+
+        //std::cout << "motor_drive: " << command.angle() << "    " << command.distance() << std::endl;
```
[the remark this answers is not in the document]
Comment (gdoffe) [reply platforms/pf-robot-motion-control/motion_control.cpp]: done

**Pull request merged**